graphene_django/relay/connection.py

```diff
@@ -77,46 +73,80 @@ def connection_from_sized_sliceable(
     else:
         raise ValueError(f"Unreachable: {args}")
 
-    first_edge_cursor: Optional[str] = edges[0].cursor if edges else None
-    last_edge_cursor: Optional[str] = edges[-1].cursor if edges else None
-
-    return connection_type(
-        edges=edges,
-        pageInfo=page_info_type(
-            startCursor=first_edge_cursor,
-            endCursor=last_edge_cursor,
-            hasPreviousPage=has_previous_page,
-            hasNextPage=has_next_page,
-        ),
-    )
+    def create_connection(handle_result: Tuple[List[Edge], bool, bool]):
+        (
+            edges,
+            has_previous_page,
+            has_next_page,
+        ) = handle_result
+
+        first_edge_cursor: Optional[str] = edges[0].cursor if edges else None
+        last_edge_cursor: Optional[str] = edges[-1].cursor if edges else None
+
+        return connection_type(
+            edges=edges,
+            pageInfo=page_info_type(
+                startCursor=first_edge_cursor,
+                endCursor=last_edge_cursor,
+                hasPreviousPage=has_previous_page,
+                hasNextPage=has_next_page,
+            ),
+        )
+
+    if GRAPHQL_SYNC_DATALOADERS_INSTALLED and isinstance(handle_result, SyncFuture):
+        return handle_result.then(create_connection)
+
+    return create_connection(handle_result)
 
 
 def _handle_no_args(
     sized_sliceable: SizedSliceable,
     edge_type,
+    info=None,
 ) -> Tuple[List, bool, bool]:
     """Handle the case where no arguments are provided."""
 
-    edges = [
-        edge_type(
-            node=node,
-            cursor=offset_to_cursor(index),
+    def compute_edges(result):
+        edges = [
+            edge_type(
+                node=node,
+                cursor=offset_to_cursor(index),
+            )
+            for index, node in enumerate(result)
+        ]
+
+        return (
+            edges,
+            False,
+            False,
         )
-        for index, node in enumerate(sized_sliceable)
-    ]
 
-    return (
-        edges,
-        False,
-        False,
-    )
+    if (
+        isinstance(sized_sliceable, QuerySet)
+        and info is not None
+        and hasattr(info, "context")
+        and hasattr(info.context, "dataloaders")
+        and isinstance(info.context.dataloaders, dict)
+    ):
+        dataloader_key = get_info_cache_key(info)
+        if dataloader_key in info.context.dataloaders:
+            return (
+                info.context.dataloaders[dataloader_key]
+                .load((sized_sliceable, None, None))
+                .then(compute_edges)
+            )
+
+        raise RuntimeError(f"Could not find dataloader for {dataloader_key}")
+
+    return compute_edges(sized_sliceable)
```

Comment:
Slight optimization proposal (to avoid calling multiple `hasattr` and dereferencing):

```suggestion
    try:
        dataloaders_context = info.context.dataloaders

    except AttributeError:
        pass

    else:
        if isinstance(dataloaders_context, dict) and isinstance(sized_sliceable, QuerySet):
            dataloader_key = get_info_cache_key(info)
            if dataloader_key in dataloaders_context:
                return (
                    dataloaders_context[dataloader_key]
                    .load((sized_sliceable, None, None))
                    .then(compute_edges)
                )

            raise RuntimeError(f"Could not find dataloader for {dataloader_key}")
        
    return compute_edges(sized_sliceable)
```